src/nvim/channel.c

```diff
@@ -754,3 +729,94 @@ static void term_close(void *data)
   multiqueue_put(chan->events, term_delayed_free, 1, data);
 }
 
+void channel_info_changed(Channel *chan, bool new)
+{
+  event_T event = new ? EVENT_CHANOPEN : EVENT_CHANINFO;
+  if (has_event(event)) {
+    channel_incref(chan);
+    multiqueue_put(main_loop.events, on_set_info_event,
+                   2, chan, event);
+  }
+}
+
+static void on_set_info_event(void **argv)
```

Comment:
For better or worse, the `*_event` handlers never are prefixed with `on_` (except in this PR and the previous channels refactor). So we should stay consistent or change all cases.

Whereas it could make sense for the `channel_info_changed` handler to be named `on_channel_info_changed`.